### PR TITLE
Remove ProblemGroup

### PR DESCRIPTION
**Pull request by reinsch82**
private #3982

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test this

**Review by hegyibalint — APPROVED** [review]

File: subprojects/problems/src/main/java/org/gradle/api/problems/BuildableProblemBuilder.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
  *          .undocumented()
  *          .noLocation()
  *          .type(ValidationProblemId.TEST_PROBLEM.name())
- *          .group(ProblemGroup.TYPE_VALIDATION)
+ *          .severity(Severity.ERROR)
```
Comment (donat) [inline subprojects/problems/src/main/java/org/gradle/api/problems/BuildableProblemBuilder.java:33]: ❌ Typo: duplication

**Review by donat — APPROVED** [review]
LGTM!

**Comment by reinsch82** [comment]
@bot-gradle test and merge

**Pull request merged**